.github/workflows/nightly.yml

```diff
@@ -22,7 +22,13 @@ jobs:
     with:
```

Comment:
why do we give the permission write here?

Reply:
it is needed by the Release job in `main.yml` (note that the actual release is skipped)

Reply:
If the release step is skipped. We don't need that permission right?

Reply:
theoretically, we don't, but the full sub-workflow needs the same permissions and that can't depend on the fact that some step is skipped somewhere later in the action. just like how in C++, a functions' `const`-ness can't depend on its arguments:

if you want to remove these permissions, the release job needs to be in a separate workflow